Build failure: super-slicer-beta

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *super-slicer-beta*

### Build log

```
nh>    Compiling clap_complete v4.4.10
nh>    Compiling subprocess v0.2.9
nh>    Compiling uzers v0.11.3
nh>    Compiling owo-colors v4.0.0
nh>    Compiling timeago v0.4.2
nh>    Compiling clean-path v0.2.1
nh>    Compiling humantime v2.1.0
nh>    Compiling nh v3.5.4 (/build/source)
nh> warning: unused import: `std::os::unix::fs::MetadataExt`
nh>   --> src/clean.rs:16:5
nh>    |
nh> 16 | use std::os::unix::fs::MetadataExt;
nh>    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
nh>    |
nh>    = note: `#[warn(unused_imports)]` on by default
nh> 
nh> warning: function `home_info` is never used
nh>    --> src/home.rs:126:4
nh>     |
nh> 126 | fn home_info() -> Result<()> {
nh>     |    ^^^^^^^^^
nh>     |
nh>     = note: `#[warn(dead_code)]` on by default
nh> 
super-slicer> make[2]: *** [src/libslic3r/CMakeFiles/libslic3r_cgal.dir/build.make:76: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/MeshBoolean.cpp.o] Error 1
super-slicer> make[1]: *** [CMakeFiles/Makefile2:1116: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/all] Error 2
super-slicer> make: *** [Makefile:156: all] Error 2
error: builder for '/nix/store/s06z0gn59i453i0q4kny91i9mj0f36hk-super-slicer-2.5.59.6.drv' failed with exit code 2;
       last 10 log lines:
       > /build/source/src/libslic3r/MeshBoolean.cpp:184:0:   required from here
       > /build/source/src/libslic3r/MeshBoolean.cpp:150: error: cannot bind non-const lvalue reference of type 'CGAL::SM_Vertex_index&' to an rvalue of type 'boost::iterators::detail::iterator_facade_base<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >::Index_iterator<CGAL::SM_Vertex_index>, CGAL::SM_Vertex_index, std::random_access_iterator_tag, CGAL::SM_Vertex_index, long int, false, false>::reference' {aka 'CGAL::SM_Vertex_index'}
       >   150 |     for (auto &vi : vertices) {
       >       |
       > /build/source/src/libslic3r/MeshBoolean.cpp:155: error: cannot bind non-const lvalue reference of type 'CGAL::SM_Face_index&' to an rvalue of type 'boost::iterators::detail::iterator_facade_base<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >::Index_iterator<CGAL::SM_Face_index>, CGAL::SM_Face_index, std::random_access_iterator_tag, CGAL::SM_Face_index, long int, false, false>::reference' {aka 'CGAL::SM_Face_index'}
       >   155 |     for (auto &face : faces) {
       >       |
       > make[2]: *** [src/libslic3r/CMakeFiles/libslic3r_cgal.dir/build.make:76: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/MeshBoolean.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:1116: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/all] Error 2
┏━ 1 Errors: ll logs, run 'nix log /nix/store/s06z0gn59i453i0q4kny91i9mj0f36hk-super-slicer-2.5.59.6.drv'.
┃ error: builder for '/nix/store/s06z0gn59i453i0q4kny91i9mj0f36hk-super-slicer-2.5.59.6.drv' failed with exit code 2;
┃        last 10 log lines:
┃        > /build/source/src/libslic3r/MeshBoolean.cpp:184:0:   required from here
error: 1 dependencies of derivation '/nix/store/mb55x2ikcx8k0l2qvyrfmpwnckfs5gcl-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vmiaxcv386g3rpdh6in8plr27sh7ak5r-nixos-system-ThinkPad-24.05.20240306.9df3e30.drv' failed to build
┏━ 1 Errors: 
┃ error: builder for '/nix/store/s06z0gn59i453i0q4kny91i9mj0f36hk-super-slicer-2.5.59.6.drv' failed with exit code 2;
┃        last 10 log lines:
┃        > /build/source/src/libslic3r/MeshBoolean.cpp:184:0:   required from here
┃        > /build/source/src/libslic3r/MeshBoolean.cpp:150: error: cannot bind non-const lvalue reference of type 'CGAL::SM_Vertex_index&' to an rvalue of type 'boost::iterators::detail::iterator_facade_base<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >::Index_iterator<CGAL::SM_Vertex_index>, CGAL::SM_Vertex_index, std::random_access_iterator_tag, CGAL::SM_Vertex_i…
┃        >   150 |     for (auto &vi : vertices) {
┃        >       |
┃        > /build/source/src/libslic3r/MeshBoolean.cpp:155: error: cannot bind non-const lvalue reference of type 'CGAL::SM_Face_index&' to an rvalue of type 'boost::iterators::detail::iterator_facade_base<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >::Index_iterator<CGAL::SM_Face_index>, CGAL::SM_Face_index, std::random_access_iterator_tag, CGAL::SM_Face_index, …
┃        >   155 |     for (auto &face : faces) {
┃        >       |
┃        > make[2]: *** [src/libslic3r/CMakeFiles/libslic3r_cgal.dir/build.make:76: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/MeshBoolean.cpp.o] Error 1
┃        > make[1]: *** [CMakeFiles/Makefile2:1116: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/all] Error 2
┃        > make: *** [Makefile:156: all] Error 2
┃        For full logs, run 'nix log /nix/store/s06z0gn59i453i0q4kny91i9mj0f36hk-super-slicer-2.5.59.6.drv'.
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
@cab404
@Moredread 
@hesiod 
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here

 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.6, NixOS, 24.05 (Uakari), 24.05.20240229.1536926`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"home-manager, nixos-23.11"`
 - nixpkgs: `/nix/store/lwyjz70qh12nq6cb7fixl85vryzxqm3c-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Having the same problem. Think it's a problem with nixos-unstable.

Logs:

```
/build/source/src/libslic3r/MeshBoolean.cpp: In instantiation of 'Slic3r::TriangleMesh Slic3r::MeshBoolean::cgal::cgal_to_triangle_mesh(const _Mesh&) [with _Mesh = CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >]>
/build/source/src/libslic3r/MeshBoolean.cpp:184:0:   required from here
/build/source/src/libslic3r/MeshBoolean.cpp:150: error: cannot bind non-const lvalue reference of type 'CGAL::SM_Vertex_index&' to an rvalue of type 'boost::iterators::detail::iterator_facade_base<CGAL::Surface_m>
  150 |     for (auto &vi : vertices) {
      | 
/build/source/src/libslic3r/MeshBoolean.cpp:155: error: cannot bind non-const lvalue reference of type 'CGAL::SM_Face_index&' to an rvalue of type 'boost::iterators::detail::iterator_facade_base<CGAL::Surface_mes>
  155 |     for (auto &face : faces) {
      | 
make[2]: *** [src/libslic3r/CMakeFiles/libslic3r_cgal.dir/build.make:76: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/MeshBoolean.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1116: src/libslic3r/CMakeFiles/libslic3r_cgal.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

Output from `nix-shell -p nix-info --run "nix-info -m"`:

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.77, NixOS, 24.05 (Uakari), 24.05pre592007.9df3e30ce24f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/misc/prusa-slicer/meshboolean-const.patch
+++ b/pkgs/applications/misc/prusa-slicer/meshboolean-const.patch
@@ -1,0 +1,19 @@
+diff --git a/src/libslic3r/MeshBoolean.cpp b/src/libslic3r/MeshBoolean.cpp
+index 31fdc35..32acbf8 100644
+--- a/src/libslic3r/MeshBoolean.cpp
++++ b/src/libslic3r/MeshBoolean.cpp
+@@ -147,12 +147,12 @@ template<class _Mesh> TriangleMesh cgal_to_triangle_mesh(const _Mesh &cgalmesh)
+     const auto &vertices = cgalmesh.vertices();
+     int vsize = int(vertices.size());
+ 
+-    for (auto &vi : vertices) {
++    for (const auto &vi : vertices) {
+         auto &v = cgalmesh.point(vi); // Don't ask...
+         its.vertices.emplace_back(to_vec3f(v));
+     }
+ 
+-    for (auto &face : faces) {
++    for (const auto &face : faces) {
+         auto vtc = cgalmesh.vertices_around_face(cgalmesh.halfedge(face));
+ 
+         int i = 0;

--- a/pkgs/applications/misc/prusa-slicer/super-slicer.nix
+++ b/pkgs/applications/misc/prusa-slicer/super-slicer.nix
@@ -16,6 +16,7 @@ let
       ];
       sha256 = "sha256-v0q2MhySayij7+qBTE5q01IOq/DyUcWnjpbzB/AV34c=";
     })
+    ./meshboolean-const.patch
   ];
 
   versions = {

